src/helpers/handlers.ts

```diff
@@ -64,7 +68,13 @@ export async function writeFileHandler(data: WriteCommandData) {
 
         // save changes
         vscode.workspace.openTextDocument(uri).then(doc => {
-            doc.save();
+            undoManager.lockDocument(doc);
+            doc.save().then(result => {
+                if (result) {
+                    undoManager.pluginFileWritten(doc);
+                }
+                undoManager.unlockDocument(doc);
+            });
```

Comment:
Not sure in what cases this promise can fail, but would it make sense to do the unlock in the finally part?:

```
doc.save()
    .then(result => {
        if (result) {
            undoManager.pluginUndoRedoPerformed(document, operation);
        }
    })
    .catch(error => {
        // Handle any error that might occur during the save or the undo operation
        console.error("Error during save operation:", error);
    })
    .finally(() => {
        undoManager.unlockDocument(doc);
    });
```

Reply:
Yes, good tip, finally seems proper place for it 🙇